### ISSUE TITLE
[Feature] visualMap-piecewise + splitnumber should support gradient color

### ISSUE THREAD

**Issue by Justin-ZS** [apache/echarts#20608]
### What problem does this feature solve?

https://github.com/apache/echarts/issues/20592
piecewise just splits values into pieces.
For each piece, I want to paint it in gradient color.
`visualMap-piecewise + pieces` can render gradient color correctly

### What does the proposed API look like?

No new API

**Comment by theo-rodrigues** [comment]
i'll handle this!

**Comment by theo-rodrigues** [comment]
how does splitNumber should use the gradient?
something like this?

splitNumber: 4,
dimension: 0,
min: 0,
max: 4,
inRange: {color: [['red', 'blue'], 'blue', 'green', ['green', 'yellow']]


should look like this?

<img width="860" height="540" alt="Image" src="https://github.com/user-attachments/assets/f9e1e8b6-525b-468e-a44e-594bc5b25da6" />

**Comment by Justin-ZS** [comment]
> how does splitNumber should use the gradient? something like this?
> 
> splitNumber: 4, dimension: 0, min: 0, max: 4, inRange: {color: [['red', 'blue'], 'blue', 'green', ['green', 'yellow']]
> 
> should look like this?
> 
> <img alt="Image" width="860" height="540" src="https://private-user-images.githubusercontent.com/140714367/475721706-f9e1e8b6-525b-468e-a44e-594bc5b25da6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ4ODUxNjQsIm5iZiI6MTc1NDg4NDg2NCwicGF0aCI6Ii8xNDA3MTQzNjcvNDc1NzIxNzA2LWY5ZTFlOGI2LTUyNWItNDY4ZS1hNDRlLTU5NGJjNWIyNWRhNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxMVQwNDAxMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMDYxMDA4MjJkYTZjOWIwZmQyY2UyYmExNmUyZGUyZmJjNjRmNWJlYjNiZjE2YmVmZWUyODJlYzg0OWQzNTBmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.jhqN8oi7WaVvvwqH6iWmEQa2xF8XVZTFiurPbxYFJUY">

My usage is set gradient color for `bar`.
```
{
    "splitNumber": 3,
    "min": 147518,
    "max": 2948082,
    "inRange": {
        "color": [
            {
                "colorStops": [
                    {
                        "offset": 0,
                        "color": "rgba(161, 188, 231, 0.3)"
                    },
                    {
                        "offset": 1,
                        "color": "rgba(161, 188, 231, 1)"
                    }
                ],
                "x": 0,
                "y": 0,
                "x2": 1,
                "y2": 0,
                "type": "linear",
                "global": false
            },
            {
                "colorStops": [
                    {
                        "offset": 0,
                        "color": "rgba(102, 146, 215, 0.3)"
                    },
                    {
                        "offset": 1,
                        "color": "rgba(102, 146, 215, 1)"
                    }
                ],
                "x": 0,
                "y": 0,
                "x2": 1,
                "y2": 0,
                "type": "linear",
                "global": false
            },
            {
                "colorStops": [
                    {
                        "offset": 0,
                        "color": "rgba(67, 121, 206, 0.3)"
                    },
                    {
                        "offset": 1,
                        "color": "rgba(67, 121, 206, 1)"
                    }
                ],
                "x": 0,
                "y": 0,
                "x2": 1,
                "y2": 0,
                "type": "linear",
                "global": false
            }
        ]
    }

}
```
Since `splitNumber` is equal to the number of colors, the actual color should be exactly one of the gradient colors.

**Comment by theo-rodrigues** [comment]
I get it, but i think i can't aplly the full gradient to the bars, maybe this can be another issue, because the ECharts does not support full gradient on series for discrete charts. Here is an example with continuous gradient using the standard ECharts configuration: 

<img width="1033" height="788" alt="Image" src="https://github.com/user-attachments/assets/72622ca0-4dc1-497e-b134-24b8129d0184" />


https://echarts.apache.org/examples/en/editor.html?c=bar-simple

``` option = {
  xAxis: {
    type: 'category',
    data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
  },
  yAxis: {
    type: 'value'
  },
  visualMap: {inRange: {color: ['red', 'green']}}
,
  series: [
    {
      data: [120, 200, 150, 80, 70, 110, 130],
      type: 'bar'
    }
  ]
};

**Comment by Justin-ZS** [comment]
> I get it, but i think i can't aplly the full gradient to the bars, maybe this can be another issue, because the ECharts does not support full gradient on series for discrete charts. Here is an example with continuous gradient using the standard ECharts configuration:
> 
> <img alt="Image" width="1033" height="788" src="https://private-user-images.githubusercontent.com/140714367/478512929-72622ca0-4dc1-497e-b134-24b8129d0184.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU1MDY2NDksIm5iZiI6MTc1NTUwNjM0OSwicGF0aCI6Ii8xNDA3MTQzNjcvNDc4NTEyOTI5LTcyNjIyY2EwLTRkYzEtNDk3ZS1iMTM0LTI0YjgxMjlkMDE4NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxOFQwODM5MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTNkZWIzZjZiM2RiMzU0NWMzNmU4NjE0NTg3NTc2Y2I5YzY3NWMwN2QzOTllNTA3MDQ2YWY1ODllZWRjZDQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HsLUYWVfzQ5DT8RVxmTNF-o76TzggQAhfrqzLAM_2aw">
> https://echarts.apache.org/examples/en/editor.html?c=bar-simple
> 
> ```
>   xAxis: {
>     type: 'category',
>     data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
>   },
>   yAxis: {
>     type: 'value'
>   },
>   visualMap: {inRange: {color: ['red', 'green']}}
> ,
>   series: [
>     {
>       data: [120, 200, 150, 80, 70, 110, 130],
>       type: 'bar'
>     }
>   ]
> };
> ```

The current behavior is correct: each bar receives a discrete color interpolated between red and green based on its value through the `visualMap`.
What I expected is individual bars with gradient fills, similar to [demo](https://echarts.apache.org/examples/zh/editor.html?c=bar-simple&code=PYBwLglsB2AEC8sDeAoWsAeBBDEDOAXMmurGAJ4gCmRA5AMYCGYVA5sAE7m0A0J6AE2aMiAbVoBZGLQC6JAL5905HPiKpSZSjVi0AbowA2AVyq0FS2HiocIVQrFH9imwcLEBGAEwAGGZc0KajoAI0YOXmd0CBYAWwBlCkMdDVd0emBDTnUo1wysjkTQByc0tNSy12AAM2rrMCIfAMrSfOzdDlYwgAoPHy8eWA8AFgA2Qa8PAFZBnwA6AGYASnMW0kVc8s2ymrqqBqHmtbaOOk6evoGhsYnpwY8V7c15J9h_V4xGo9dyL4-vP5rcgAw6vII6WiGCDQKjhSJrVhZMKGIjVIzWbYvNJY9YkOTyADcQA)

Simply setting gradient colors in `inRange.color` may not be the most general solution. 
Perhaps we need a new property to transform discrete colors into gradient fills:
```
inRange: {
    color: ['red', 'green'],
    colorTransform: // a function to generate the final gradient color
}
```
@Ovilia  @100pah do you have any thoughts on this approach?

**Comment by theo-rodrigues** [comment]
To make the contribution easier to manage and the pull requests simpler to review, maybe we can break this feature down into smaller parts. I can create a separate issue for implementing the basic gradient support first, submit a PR for that, and then create another to deal with the gradient per series for discrete charts. This approach seems much more simpler and focused.

Would that be a good approach for you?

**Comment by Ovilia** [comment]
First, let's make it clear: why is there a need to use gradient colors in a piecewise visualMap? In my opinion, such a design may increase the user's comprehension cost and even lead to misunderstandings. Colors sometimes represent mapping relationships, while the gradient color part may indicate the same value. This logic seems quite confusing to me. Perhaps using a continuous visualMap would be more reasonable in such cases.

**Comment by theo-rodrigues** [comment]
I understand @Ovilia , and thanks for joining the discussion, i'd like to try to answer your question. The use case that i'm dealing with is a chart that acts like a health monitor, above the threshold the line should be a gradient, and below should also be another gradient, something like that: 

<img width="807" height="442" alt="Image" src="https://github.com/user-attachments/assets/05f38238-ab3d-4cc7-81fc-24bfa2eda2a1" />


Other example:

<img width="791" height="394" alt="Image" src="https://github.com/user-attachments/assets/bb055fee-484d-4666-b77f-b90a53ce2004" />

That example can be replicated with continuous, but it is much more complex since i have to decrease the min to make the right color configuration, and even so, there is a discrepancy between the colors.

<img width="852" height="527" alt="Image" src="https://github.com/user-attachments/assets/effd07db-da56-46c9-a765-8d55bd55e5fd" />

And with piecewise is much simpler, i just have to set the interval and make the gradients.


In summary, I think that this contribution will improve the level of customization with interval to another level, i can mixed up solid colors and gradient with piecewise, can select or not the piece with gradient, with piecewise the manage is more accurate.

thanks again to joining and trying to understand the problem.

**Comment by theo-rodrigues** [comment]
Hi @Ovilia ,

I'm checking in on this issue to see if there are any updates or next steps. As I mentioned, the original author is ready to break down the proposal into smaller pieces.

Please let me know if there's anything I can do to help, such as creating a draft of the smaller issues or providing more details.

Thanks!

**Comment by PaulKreisel** [comment]
I just created the issue https://github.com/apache/echarts/issues/21437 which is a partial duplicate of this issue. I also need to be able to give stop values for continuous visual maps.
